orangecontrib/spectroscopy/widgets/owspectra.py

```diff
@@ -809,6 +811,46 @@ def __init__(self, parent: OWWidget, select=SELECTNONE):
         )
         self.invertX_menu.setShortcutContext(Qt.WidgetWithChildrenShortcut)
         actions.append(self.invertX_menu)
+
+        peaklabler_menu = MenuFocus("Peak Labeling Options", self)
+        peak_action = QWidgetAction(self)
+        layout = QGridLayout()
+        prominence_box = gui.widgetBox(self, margin=5, orientation=layout)
+        prominence_box.setFocusPolicy(Qt.TabFocus)
+        self.single_peak = QAction(
+            None, self, shortcut=Qt.Key_P, checkable=False,
+            triggered=self.peak_apply
+        )
+        self.single_peak.setShortcutContext(Qt.WidgetWithChildrenShortcut)
+        self.single_peak = QPushButton("Add Single Line", self)
+        self.reset_labels = QPushButton('Reset graph', self)
+        self.single_peak.clicked.connect(self.find_peak_variables)
+        self.single_peak.clicked.connect(self.peak_apply)
+        self.reset_labels.clicked.connect(self.change_state)
```

Comment:
`change_state` should be removed. Peak updating needs to be handled differently.